documentation/library_guide/kernel_templates/kernel_configuration.rst

```diff
@@ -0,0 +1,39 @@
+Kernel Configuration
+####################
+
+.. code:: cpp
+
+   // Defined in header <oneapi/dpl/experimental/kernel_templates>
+
+   template <std::uint16_t DataPerWorkItem,
+             std::uint16_t WorkGroupSize,
+             typename KernelName = oneapi::dpl::execution::DefaultKernelName>
+   struct kernel_param;
+
+Each Kernel Template is supplied by a `kernel_param` object, generic structure for configuring a kernel.
+The supported and the most performant values depend on the Kernel Template itself, device capabilities, the data type and the data size.
+
+Member constants
+----------------
+
++------------------------------------------------------+---------------------+------------------------------------------------------+
+| Name                                                 | Value               | Description                                          |
++======================================================+=====================+======================================================+
+| ``static constexpr std::uint16_t data_per_workitem`` | ``DataPerWorkItem`` | Number of iterations to be processed by a work-item. |
++------------------------------------------------------+---------------------+------------------------------------------------------+
+| ``static constexpr std::uint16_t workgroup_size``    | ``WorkGroupSize``   | Number of work-items within a work-group.            |
++------------------------------------------------------+---------------------+------------------------------------------------------+
+
+
+Member types
+------------
+
++-----------------+----------------+----------------------------------------------------------------------------------------+
+| Type            | Definition     | Description                                                                            |
```

Comment:
For the description, my suggestion:

Name of the kernel. It can be augmented for a ``kernel_param`` object passed in a multi-kernel template to guarantee the uniqueness of the individual kernels. For ``DefaultKernelName``, a unique implementation-defined name is generated across multiple kernel templates if ``fsycl-unnamed-lambda`` is enabled.

Reply:
I have described ``kernel_name`` more for clarity. Could you check it once again? I apologise for making the suggestion not relevant.

Reply:
Thank you for the update. My suggestions:

An optional parameter that is used to set a kernel name. The behavior is different whether the parameter is provided or not:
- Provided: It is passed as is or augmented to guarantee unique kernel names in the kernel template if it has multiple kernels.
- Not provided: The default value (DefaultKernelName) results in the implementation-defined generation of unique kernel names across the whole program.

Note::

Automatic kernel name generation (the **not provided** option) assumes that the used compiler and runtime comply with SYCL* 2020 and support optional kernel names. For example, the Intel® oneAPI DPC++/C++ Compiler supports it by default.